src/main/java/org/sagebionetworks/bridge/participantroster/DownloadParticipantRosterWorkerProcessor.java

```diff
@@ -0,0 +1,198 @@
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * Worker used to download all participants into a CSV and email it to researchers.
+ */
+@Component("DownloadParticipantRosterWorker")
+public class DownloadParticipantRosterWorkerProcessor implements ThrowingConsumer<JsonNode> {
+    private static final Logger LOG = LoggerFactory.getLogger(DownloadParticipantRosterWorkerProcessor.class);
+
+    // If there are a lot of downloads, write log messages regularly so we know the worker is still running.
+    private static final int REPORTING_INTERVAL = 100;
+
+    static final int PAGE_SIZE = 100;
+    private static final long THREAD_SLEEP_INTERVAL = 1000L;
+
+    private BridgeHelper bridgeHelper;
+    private FileHelper fileHelper;
+    private SesHelper sesHelper;
+    private DynamoHelper dynamoHelper;
+    private ZipHelper zipHelper;
+
+    /** Helps call Bridge Server APIs */
+    @Autowired
+    public final void setBridgeHelper(BridgeHelper bridgeHelper) {
+        this.bridgeHelper = bridgeHelper;
+    }
+
+    /** Wrapper class around the file system. Used by unit tests to test the functionality without hitting the real file
+     * system.
+     */
+    @Autowired
+    public final void setFileHelper(FileHelper fileHelper) {
+        this.fileHelper = fileHelper;
+    }
+
+    /** SES helper, used to email the pre-signed URL to the requesting user. */
+    @Autowired
+    public final void setSesHelper(SesHelper sesHelper) {
+        this.sesHelper = sesHelper;
+    }
+
+    /** Dynamo helper*/
+    @Autowired
+    public final void setDynamoHelper(DynamoHelper dynamoHelper) {
+        this.dynamoHelper = dynamoHelper;
+    }
+
+    @Autowired
+    public final void setZipHelper(ZipHelper zipHelper) {
+        this.zipHelper = zipHelper;
+    }
+
+    /** Main entry point into Download Participant Roster Worker. */
+    @Override
+    public void accept(JsonNode jsonNode) throws Exception {
+
+        BridgeDownloadParticipantRosterRequest request;
+        try {
+            request = DefaultObjectMapper.INSTANCE.treeToValue(jsonNode, BridgeDownloadParticipantRosterRequest.class);
+        } catch (IOException e) {
+            throw new PollSqsWorkerBadRequestException("Error parsing request: " + e.getMessage(), e);
+        }
+
+        String userId = request.getUserId();
+        String appId = request.getAppId();
+        String password = request.getPassword();
+        LOG.info("Received request for userId=" + userId + ", app=" + appId + ", password=" + password);
+
+        Stopwatch requestStopwatch = Stopwatch.createStarted();
+        File csvFile = null;
+        File tmpDir = fileHelper.createTempDir();
+        File zipFile = null;
+
+        try {
+            // get caller's user using Bridge API getParticipantByIdForApp
+            StudyParticipant participant = bridgeHelper.getParticipant(appId, userId, false);
+            String orgMembership = participant.getOrgMembership();
```

Comment:
Should probably add a quick sanity check here that the participant has an email address. This will ensure we don't spend a lot of work downloading a roster just to throw because we have a null email address.

Also, we should check that emailVerified is true. This way, we don't accidentally spam somebody who didn't verify their email address.

If both of those checks fail, the cleanest thing to do is log a warning and return.